src/Data/Binary/Class.hs

```diff
@@ -2,6 +2,10 @@
 {-# LANGUAGE DefaultSignatures #-}
 {-# LANGUAGE Safe #-}
 
+#if __GLASGOW_HASKELL__ >= 706
+{-# LANGUAGE PolyKinds #-}
+#endif
```

Comment:
This is needed to ensure that the `Binary (Alt f a)` instance is generalized to be of kind `Alt (f :: k -> *) (a :: k)` instead of kind `Alt (f :: * -> *) (a :: *)`.